### PR TITLE
Implement LengthPrefixCoder.to_type_hint

### PR DESCRIPTION
**Pull request by hjtran**
LengthPrefixCoder doesn't define `to_type_hint` even though it seems pretty well defined. I've implemented it by just inferring it from its value coder.

Motivating use case - we have some debug tools that infer the types of pcollections based on `to_type_hint`. The LPCoder is used in a lot of places.

Bonus/unrelated change - I've pulled out the GitHub issue referenced in the `NotImplementedError` raised by `to_type_hint`. Every time I run into this error, I check the github issue and get more confused. It doesn't seem useful to refer to it in the error message, but does still seem relevant, so I pulled it out into a comment.

*Testing Done*
I haven't actually run the new unit test I wrote since I'm struggling to get my python sdk dev environment up (I've posted on the dev@ mailing list). Hoping the GH Action can just do it for me :)

### CODE REVIEW

**Comment by hjtran** [comment]
R: @robertwb

**Comment by hjtran** [comment]
Bump @robertwb (lmk if I should just use the regular reviewer-assignment system going forward)

**Comment by hjtran** [comment]
Not sure who to @ here to keep this moving. @damccorm ?

File: local-env-setup.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ darwin_install_pip3_packages() {
 
 install_go_packages(){
         echo "Installing goavro"
-        go get github.com/linkedin/goavro/v2
+        go install github.com/linkedin/goavro@latest
```
Comment (damccorm) [inline local-env-setup.sh:27]: Did you mean to make this change? I think we need get here not just install

**Review by damccorm — COMMENTED** [review]
Thanks for this, and sorry I'm behind on reviews - the core change LGTM, I just had a question about the go change

File: local-env-setup.sh (hunk 1 above)
Comment (hjtran) [reply local-env-setup.sh:27]: No, this was accidental. Had to make this change to get my dev container to work. [Mailing list thread](https://lists.apache.org/thread/4lp8nqppd2xv8m694oyjzzjqvwtor4gl)

Will revert

**Review by damccorm — APPROVED** [review]
Thanks!

**Comment by damccorm** [comment]
I'll merge later after checks pass

**Comment by hjtran** [comment]
> I'll merge later after checks pass

Thanks for all the reviews! I really appreciate it!

**Comment by damccorm** [comment]
Uh oh, looks like a bad rebase

**Comment by hjtran** [comment]
I've just created a duplicate PR without the rebase issues
https://github.com/apache/beam/pull/30857

**Comment by damccorm** [comment]
SGTM

**Pull request closed without merging**